airbyte-cdk/python/airbyte_cdk/sources/declarative/incremental/per_partition_cursor.py

```diff
@@ -107,13 +107,14 @@ def __init__(self, cursor_factory, partition_router):
         self._cursor_factory = cursor_factory
         self._partition_router = partition_router
         self._cursor_per_partition = {}
+        self._default_cursor_state = self._NO_CURSOR_STATE
 
     def stream_slices(self, sync_mode: SyncMode, stream_state: StreamState) -> Iterable[PerPartitionStreamSlice]:
         slices = self._partition_router.stream_slices(sync_mode, self._NO_STATE)
         for partition in slices:
             cursor = self._cursor_per_partition.get(self._to_tuple(partition))
             if not cursor:
-                cursor = self._create_cursor(self._NO_CURSOR_STATE)
+                cursor = self._create_cursor(self._default_cursor_state)
```

Comment:
For DatetimeBasedCursor, this is useless because it only relies on the `stream_state` passed in arguments to `stream_slices`. However, there might be custom components that relies on their internal state instead of this parameter. This problem comes from the fact that we have two different ways to pass state to stream slicers and therefore, there is an uncertainty as to which one the implementation will us